File: java/src/main/java/leetcode/ExploreInsertIntoABinarySearchTree.java
package leetcode;

import leetcode.api.TreeNode;

/**
 * https://leetcode.com/explore/learn/card/introduction-to-data-structure-binary-search-tree/141/basic-operations-in-a-bst/1003/
 */
public final class ExploreInsertIntoABinarySearchTree {
    public static TreeNode insertIntoBST(TreeNode root, int val) {
        if (root == null) {
            return new TreeNode(val);
        }

        if (val > root.val) {
            root.right = insertIntoBST(root.right, val);
        } else {
            root.left = insertIntoBST(root.left, val);
        }

        return root;
    }
}

File: kotlin/src/main/kotlin/leetcode/ExploreInsertIntoABinarySearchTree.kt
package leetcode

import leetcode.api.TreeNode

/**
 * https://leetcode.com/explore/learn/card/introduction-to-data-structure-binary-search-tree/141/basic-operations-in-a-bst/1003/
 */
class ExploreInsertIntoABinarySearchTree {
    fun insertIntoBST(root: TreeNode? = null, value: Int): TreeNode {
        if (root == null) {
            return TreeNode(value)
        }

        if (value > root.value) {
            root.right = insertIntoBST(root.right, value)
        } else {
            root.left = insertIntoBST(root.left, value)
        }

        return root
    }
}
